### PR TITLE
HDS-1512-file-input-default-value-prop

### PR DESCRIPTION
**Pull request by mrTuomoK**
## Description
Add defaultValue prop to FileInput, add corresponding documentations & demos.

## Related Issue
[HDS-1512](https://helsinkisolutionoffice.atlassian.net/browse/HDS-1512)

## Motivation and Context
For example when using a multi-page form you might want to remember the chosen files later on, not emptying the FileInput.

## How Has This Been Tested?
- Local machine with multiple browsers (Safari, Firefox, Chrome), on iOS-devices
- Also created visual tests for this
- Jest-tests were left undone since mocking a needed FileList-object was not resolved (yet eat least)

[HDS-1512]: https://helsinkisolutionoffice.atlassian.net/browse/HDS-1512?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: site/src/docs/components/file-input/index.mdx [hunk 1]
```diff
@@ -96,3 +96,28 @@ export const FileInputDragAndDrop = () => {
...
   <FileInputDragAndDrop />
 </PlaygroundPreview>
+
+#### With default value
+
+If needed, the file input can be prefilled with pre-existing file(s). Useful, for example, if adding files in a multi-page form and coming back to the file-adding step again.
+
+export const FileInputDefaultValue = () => {
+  const [files, setFiles] = React.useState([
+    new File(['string content'], 'dummy.txt', { type: 'text/plain' }),
```
Comment (mikkojamG) [inline site/src/docs/components/file-input/index.mdx:106]: Causes error on documentation gatsby build `WebpackError: ReferenceError: File is not defined` 🤔

File: packages/react/src/components/fileInput/FileInput.stories.tsx [hunk 2]
```diff
@@ -67,6 +67,35 @@ Multiple.args = {
...
+    <FileInput
+      multiple={args.multiple}
+      defaultValue={args.defaultValue}
+      id={args.id}
+      label={args.label}
+      buttonLabel={args.buttonLabel}
+      language={args.language}
+      onChange={setFiles}
+      maxSize={args.maxSize}
+      accept={args.accept}
```
[the remark this answers is not in the document]
Comment (harriplappalainen) [reply packages/react/src/components/fileInput/FileInput.stories.tsx:85]: This default value could be put in the Stepper story as well.

**Review by harriplappalainen — COMMENTED** [review]
Against all the scepticism this feature faced beforehand, it works without a hitch! 💪 I had just minor comments.

File: packages/react/src/components/fileInput/FileInput.stories.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (harriplappalainen) [reply packages/react/src/components/fileInput/FileInput.stories.tsx:85]: The stepper story SimpleFormExample was so complex that I had to test myself how to get the default value FileInput working there. Now that it works and you're currently sick, I hope it's not weird that I'll actually push the changes I requested myself 😅

File: packages/react/src/components/fileInput/FileInput.tsx [hunk 3]
```diff
@@ -25,6 +25,10 @@ type FileInputProps = {
    * Additional class names to apply to the file input
    */
   className?: string;
+  /**
+   * Default selected files for the input
```
[the remark this answers is not in the document]
Comment (harriplappalainen) [reply packages/react/src/components/fileInput/FileInput.tsx:29]: Although the value is an array so maybe it's ok to have it as `files`.

File: packages/react/src/components/fileInput/FileInput.stories.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (mrTuomoK) [reply packages/react/src/components/fileInput/FileInput.stories.tsx]: Be my guest, good comments & changes 👌🏻

[the remark this answers is not in the document]
Comment (harriplappalainen) [reply packages/react/src/components/fileInput/FileInput.stories.tsx]: Yes, if I could only change things without breaking [every test](https://github.com/City-of-Helsinki/helsinki-design-system/actions/runs/4362391061/jobs/7627439021) 🤣 I'll handle the mess I caused.

File: site/src/docs/components/file-input/index.mdx (hunk 1 above)
Comment (mrTuomoK) [reply site/src/docs/components/file-input/index.mdx]: Ah yeah the SSR 🤦

Comment (mikkojamG) [reply site/src/docs/components/file-input/index.mdx]: This appears to be fixed, resolving 👍

File: packages/react/src/components/fileInput/FileInput.tsx [hunk 4]
```diff
@@ -530,6 +534,26 @@ export const FileInput = ({
...
+
+      defaultValue.forEach((defaultFile) => {
+        const file = new File([defaultFile], defaultFile.name, {
+          type: defaultFile.type,
+          lastModified: defaultFile.lastModified,
+        });
+        dataTransfer.items.add(file);
+      });
+
+      if (inputRef.current) {
```
Comment (mikkojamG) [inline packages/react/src/components/fileInput/FileInput.tsx:549]: I think you could move this to check the upper if statement 🤔

File: packages/react/src/components/fileInput/FileInput.stories.tsx [hunk 5]
```diff
@@ -24,17 +24,7 @@ export const Single = (args) => {
   // eslint-disable-next-line no-console
   console.log('selected file', file);
```
Comment (tuomashatakka) [inline packages/react/src/components/fileInput/FileInput.stories.tsx:25]: Would it make sense to use a storybook action here instead? :)

**Review by tuomashatakka — CHANGES_REQUESTED** [review]

**Review by tuomashatakka — APPROVED** [review]

**Pull request merged**